### ISSUE TITLE
Support typed null pushdown in QueryService

### ISSUE THREAD

**Issue by vitalyisaev2** [ydb-platform/ydb-go-sdk#1506]
# Feature Request

**Describe the Feature Request**

Consider we have the following table in YDB:

```sql
    CREATE TABLE pushdown (
        id Int32 NOT NULL,
        col_01_int Int32,
        col_02_text UTF8,
        PRIMARY KEY (id) 
    );
    INSERT INTO pushdown (id, col_01_int, col_02_text) VALUES
        (1, 10, "a"),
        (2, 20, "b"),
        (3, 30, "c"),
        (4, NULL, NULL);
```

We need to execute SQL query with NULL value  `=` comparison (yes I know it's weird query and it always returns empty result; see YQ-3711 for details):

```sql
SELECT * FROM pushdown WHERE col_01_int = NULL
```

This query worked fine via `database/sql` driver (this is a pseudocode below, but it precisely reflects the main idea):

```sql
db.Query(ctx, "SELECT * FROM pushdown WHERE `col_01_int` = ?", (*int32)(nil))
```

But it's not clear how to express the same query using Native SDK and Query Service. 

```go
err := db.Query().Do( 
   ctx, 
   func(ctx context.Context, s query.Session) (err error) { // retry operation
        queryText := `
        DECLARE $p0 AS Int32;
        SELECT * FROM pushdown WHERE `col_01_int` = $p0;
        `
   	streamResult, err := s.Query(
              ctx, 
              queryText,
              // What method to use to pass typed null value?
              query.WithParameters(ydb.ParamsBuilder().Param('col_01_int').???.Build())) 
        )
```

Please provide some examples of filter pushdown via Query Service when some args are typed nulls. This is a blocker for YQ-3730.

**Comment by asmyasnikov** [comment]
Use
ydb.ParamsBuilder().Param('col_01_int').BeginOptional(). Uint32(nil). EndOptional().Build()))

**Comment by vitalyisaev2** [comment]
@asmyasnikov thanks, unfortunally this didn't help.

The table:
```sql
    CREATE TABLE simple (id Int32 NOT NULL, col1 String, col2 Int32, PRIMARY KEY (id));
    COMMIT;
    INSERT INTO simple (id, col1, col2) VALUES
      (1, "ydb_a", 10),
      (2, "ydb_b", 20),
      (3, "ydb_c", 30),
      (4, "ydb_d", 40),
      (5, "ydb_e", 50),
      (6, "ydb_g", NULL);
    COMMIT;
```

The code:
```go
		queryText := `
		DECLARE $p0 AS Optional<Int32>;
		SELECT * FROM %s WHERE col2 = $p0;
		`

		paramsBuilder := ydb.ParamsBuilder()
		paramsBuilder = paramsBuilder.Param("$p0").BeginOptional().Int32(nil).EndOptional()

		result, err := s.Query(ctx, fmt.Sprintf(queryText, tableName), query.WithParameters(paramsBuilder.Build()))
```

This results in error:
```
pool.With failed with 1 attempts: non-retryable error occurred on attempt No.1 (idempotent=false): query error: operation/BAD_REQUEST (code = 400010, address = localhost:2136, issues = [{'ydb/core/kqp/session_actor/kqp_session_actor.cpp:1038: ydb/core/kqp/query_data/kqp_query_data.cpp:277: Parameter $p0 type mismatch, expected: { Kind: Optional Optional { Item { Kind: Data Data { Scheme: 1 } } } }, actual: Type (Optional) { Optional item type: { Type (Optional) { Optional item type: { Type (Data), schemeType: Int32, schemeTypeId: 1 } } } } '}])
```

Steps to reproduce:
```bash
git clone https://github.com/ydb-platform/fq-connector-go
cd tools/ydb/query_service_negative
docker-compose up -d #wait for a while to let YDB initialize
go run .
```

**Comment by vitalyisaev2** [comment]
Reproducing this in integration test: https://github.com/ydb-platform/ydb-go-sdk/blob/YQ-3730.test/tests/integration/query_execute_test.go#L414-L472

**Comment by rekby** [comment]
The bug was fixed in v3.89.0, check it on your side please.
